crates/swc_ecma_transforms_base/src/resolver/mod.rs

```diff
@@ -418,7 +423,20 @@ impl<'a> VisitMut for Resolver<'a> {
         // We need to analyze these identifiers for type stripping purposes.
         self.ident_type = IdentType::Binding;
         self.in_type = n.type_only;
+
         n.visit_mut_children_with(self);
+
+        for specifier in &n.specifiers {
+            match specifier {
+                ImportSpecifier::Default(ImportDefaultSpecifier { ref local, .. })
+                | ImportSpecifier::Named(ImportNamedSpecifier { ref local, .. })
```

Comment:
I think we also have to check for `type` of `ImportNamedSpecifier`.

```ts
import { type foo, bar } from 'baz';

```

Reply:
Also, it would be nice to have a test case for the import namespace specifier with `type` keywortd.

Reply:
I may misunderstood but `import { type ..}` seems just works? tried to add test cases here

https://github.com/swc-project/swc/pull/3130/files#diff-92b6d0b97f5a9184ecd099b4e07cf97b45aef1222a7da1d40f891ec385d8fc83R2